tests/kubemark/build-kubemark.sh

```diff
@@ -0,0 +1,271 @@
+        ;;
+    --kubemark-cluster-template-url)
+        if [ -n "$2" ]; then
+            KUBEMARK_CLUSTER_TEMPLATE_URL="$2"
+            shift
+        fi
+        ;;
+    --external-cluster-template-url)
+        if [ -n "$2" ]; then
+            EXTERNAL_CLUSTER_TEMPLATE_URL="$2"
+            shift
+        fi
+        ;;
+    --hollow-nodes-deployment-url)
+        if [ -n "$2" ]; then
+            HOLLOW_NODES_DEPLOYMENT_URL="$2"
+            shift
+        fi
+        ;;
+    --clusterloader2-bin-url)
+        if [ -n "$2" ]; then
+            CLUSTERLOADER2_BIN_URL="$2"
+            shift
+        fi
+        ;;
+    *)  
+        echo "$1 is not a supported option"
+        exit 99
+        ;;  
+  esac  
+  shift  
+done
+
+psd="/proc/sys/kernel/random/uuid"
+uuid=$(cat $psd)
+prefix=${uuid:0:4}
+
+KUBEMARK_CLUSTER_RESOURCE_GROUP="${KUBEMARK_CLUSTER_RESOURCE_GROUP:-kubemark-cluster-$prefix}"
+EXTERNAL_CLUSTER_RESOURCE_GROUP="${EXTERNAL_CLUSTER_RESOURCE_GROUP:-kubemark-external-cluster-$prefix}"
+
+LOCATION="${LOCATION:-southcentralus}"
+
+KUBEMARK_SIZE="${KUBEMARK_SIZE:-100}"
+
+PRIVATE_KEY="${PRIVATE_KEY:-${HOME}/.ssh/id_rsa}"
+PUBLIC_KEY="${PRIVATE_KEY:-${HOME}/.ssh/id_rsa_pub}"
+
+# install azure cli
+curl -sL https://aka.ms/InstallAzureCLIDeb | bash
+
+# read azure credentials
+ClientID=$(jq '.ClientID' "${AZURE_CREDENTIALS}" | sed 's/"//g')
+ClientSecret=$(jq '.ClientSecret' "${AZURE_CREDENTIALS}" | sed 's/"//g')
+TenantID=$(jq '.TenantID' "${AZURE_CREDENTIALS}" | sed 's/"//g')
+
+echo "got azure credentials: ClientID ($ClientID), ClientSecret ($ClientSecret), TenantID ($TenantID)"
+
+az login --service-principal --username "${ClientID}" --password "${ClientSecret}" --tenant "${TenantID}"
+
+AKS_ENGINE="${WORKING_DIR}/aks-engine"
+
+function ssh_and_do {
+    if [ -f "$2" ]; then
+        ssh -o StrictHostKeyChecking=no -i "${PRIVATE_KEY}" kubernetes@"$1" < "$2"
+    else
+        ssh -o StrictHostKeyChecking=no -i "${PRIVATE_KEY}" kubernetes@"$1" "$2"
+    fi
+}
+
+function create_resource_group {
+    az group create -n "$1" -l "${LOCATION}" --tags "autostop=no"
+}
+
+function cleanup {
+    echo "cleaning up resource groups..."
+
+    az group delete -n "${KUBEMARK_CLUSTER_RESOURCE_GROUP}" -y --no-wait
+    az group delete -n "${EXTERNAL_CLUSTER_RESOURCE_GROUP}" -y --no-wait
+
+    az logout
+}
+
+trap cleanup ERR EXIT
+
+function get_master_ip {
+    KUBEMARK_MASTER_IP=$(az network public-ip list -g "$1" | jq '.[0].ipAddress' | sed 's/"//g')
+}
+
+function build_kubemark_cluster {
+    ${AKS_ENGINE} generate "$1"
+
+    KUBEMARK_CLUSTER_DNS_PREFIX=$(jq '.properties.masterProfile.dnsPrefix' "$1" | sed 's/"//g')
+    az group deployment create \
+      -g "${KUBEMARK_CLUSTER_RESOURCE_GROUP}" \
+      --template-file "${WORKING_DIR}/_output/${KUBEMARK_CLUSTER_DNS_PREFIX}/azuredeploy.json" \
+      --parameters "${WORKING_DIR}/_output/${KUBEMARK_CLUSTER_DNS_PREFIX}/azuredeploy.parameters.json"
+
+    echo '#! /bin/bash
```

Comment:
could we move these lines to a file? drawback is it needs another file to download